apps/simple-server-example/src/client/client.tsx

```diff
@@ -0,0 +1,92 @@
+import { useMultiplayerSync } from '@tldraw/sync'
+import { useCallback } from 'react'
+import { createRoot } from 'react-dom/client'
+import {
+	Route,
+	RouterProvider,
+	createBrowserRouter,
+	createRoutesFromElements,
+	useParams,
+} from 'react-router-dom'
+import { AssetRecordType, Editor, TLAsset, TLAssetStore, Tldraw, getHashForString } from 'tldraw'
+import 'tldraw/tldraw.css'
+
+const assets: TLAssetStore = {
+	resolve: async (asset) => {
+		return asset.props.src
+	},
+	upload: async (asset, file) => {
+		const url = `http://localhost:5858/assets/${encodeURIComponent(asset.id)}`
+		const res = await fetch(url, { method: 'PUT', body: file })
+		if (!res.ok) {
+			throw new Error('Failed to upload asset')
+		}
+		return url
+	},
+}
+
+export async function unfurl({ url }: { url: string }): Promise<TLAsset> {
+	const urlHash = getHashForString(url)
+	let meta: { description: string; image: string; favicon: string; title: string } | undefined
+	try {
+		meta = (await (
+			await fetch(`http://localhost:5858/unfurl?url=${encodeURIComponent(url)}`)
+		).json()) as any
+	} catch (e) {
+		console.error(e)
+	}
+
+	return {
+		id: AssetRecordType.createId(urlHash),
+		typeName: 'asset',
+		type: 'bookmark',
+		props: {
+			src: url,
+			description: meta?.description ?? '',
+			image: meta?.image ?? '',
+			favicon: meta?.favicon ?? '',
+			title: meta?.title ?? '',
+		},
+		meta: {},
+	}
+}
+
+function App() {
+	const { roomId } = useParams()
+	const store = useMultiplayerSync({
+		uri: 'ws://localhost:5858/connect/' + roomId,
+		assets,
+		onEditorMount: useCallback((editor: Editor) => {
+			editor.registerExternalAssetHandler('url', unfurl)
+		}, []),
```

Comment:
unless we have a good reason to, lets stick with the normal component `onMount` callback. it means we can skip the `useCallback` and will be more familiar wrt the other examples

Reply:
copied your client code